### PR TITLE
Update logging.rb

### PR DESCRIPTION
**Pull request by henryzhang87**
including the json logging format

## Context

Please describe any relevant motivation and context for these changes.

Related to # (issue)

## What's New

Please summarize what's new or changed in this PR.

Remember to request a reviewer when you submit!

### CODE REVIEW

File: lib/pushmi_pullyu/logging.rb [hunk 1]
```diff
@@ -72,8 +73,41 @@ def log_preservation_event(deposited_file, aip_directory)
...
+           tmp_hash["fileset_uuid"]= "#{file_detail[:fileset_name]}"
+           tmp_hash["file_name"]= "#{file_detail[:file_name]}"
+           tmp_hash["file_type"]= "#{file_detail[:file_extension]}"
+           tmp_hash["file_size"]= "#{file_detail[:file_size]}"
+           tmp_details << tmp_hash
+        end
+      end
+
+      message["aip_file_details"]=tmp_details
+      return message.to_json
```
[the remark this answers is not in the document]
Comment (pgwillia) [reply lib/pushmi_pullyu/logging.rb]: You indicated that you weren't sure about this change so here's more explanation about why it's equivalent. In ruby, [by default, a method returns the last expression that was evaluated in the body of the method](https://docs.ruby-lang.org/en/2.7.0/syntax/methods_rdoc.html#label-Return+Values).  So unless you need to return early (a technique called a guard clause) there isn't a need for return at the end of the method.  The Ruby Style Guide suggests that [no explicit return](https://rubystyle.guide/#no-explicit-return) is preferred.

**Review by pgwillia — APPROVED** [review]
This looks good!  There are two more nags from the houndci-bot about style that you could address.  The only thing I might suggest to improve it is to add a comment to the `preservation_event_to_json` method just explaining why and maybe giving an example of the output you'd expect.  Something like

```
# In case we need to validate or audit what is preserved into Swift, it is hard to parse the original log format to get information. 
# With the json format, (even though it has an annoying prefix, but the prefix can be removed with a single bash command), it is a lot easier to parse.
# example output: 
#  I, [2022-04-06T11:07:21.983875 #20791]  INFO -- : {"do_uuid":"2dfb153e-76da-4cac-a968-a3f6b2e3a61c","aip_deposited_at":"Wed, 06 Apr 2022 17:07:22 GMT","aip_md5sum":"8b2e08e4c3c72054b8d0ecb8c3a52ee8","aip_sha256":"","aip_metadata":"{\"project-id\":\"2dfb153e-76da-4cac-a968-a3f6b2e3a61c\",\"aip-version\":\"1.0\",\"project\":\"ERA\",\"promise\":\"bronze\"}","aip_file_details":[{"fileset_uuid":"063395c0-c947-438b-b154-ff1ac8baddbd","file_name":"ArchivingKnowledge_RepatBooklet.pdf","file_type":"pdf","file_size":"5189798"}]},
def preservation_event_to_json(deposited_file, aip_directory)
```
```

**Comment by henryzhang87** [comment]
> This looks good! There are two more nags from the houndci-bot about style that you could address. The only thing I might suggest to improve it is to add a comment to the `preservation_event_to_json` method just explaining why and maybe giving an example of the output you'd expect. Something like
> 
> ```
> # In case we need to validate or audit what is preserved into Swift, it is hard to parse the original log format to get information. 
> # With the json format, (even though it has an annoying prefix, but the prefix can be removed with a single bash command), it is a lot easier to parse.
> # example output: 
> #  I, [2022-04-06T11:07:21.983875 #20791]  INFO -- : {"do_uuid":"2dfb153e-76da-4cac-a968-a3f6b2e3a61c","aip_deposited_at":"Wed, 06 Apr 2022 17:07:22 GMT","aip_md5sum":"8b2e08e4c3c72054b8d0ecb8c3a52ee8","aip_sha256":"","aip_metadata":"{\"project-id\":\"2dfb153e-76da-4cac-a968-a3f6b2e3a61c\",\"aip-version\":\"1.0\",\"project\":\"ERA\",\"promise\":\"bronze\"}","aip_file_details":[{"fileset_uuid":"063395c0-c947-438b-b154-ff1ac8baddbd","file_name":"ArchivingKnowledge_RepatBooklet.pdf","file_type":"pdf","file_size":"5189798"}]},
> def preservation_event_to_json(deposited_file, aip_directory)
> ```

Comments added to the method "preservation_event_to_json" in henryzhang87:patch-1.

**Review by pgwillia — APPROVED** [review]
Looks good!  When you're ready go ahead and click the button on [this page](https://github.com/ualbertalib/pushmi_pullyu/pull/246) to merge.

image

**Comment by henryzhang87** [comment]
@pgwillia when it is convenient for you, could you grant me write access to the repository to merge the pull requests? Or if it is not a good idea to grant me write permission to the repository, could someone merge the pull requests on my behalf?

Also, the gems need to be published to rubygems.org after the merge, if my understanding is not wrong.

**Comment by pgwillia** [comment]
Sorry for not realizing that you needed authorization!  You should have it now.  I can publish the rubygem, or do you want experience with that part of the process too?  I'd be happy to walk you through it.

> @pgwillia when it is convenient for you, could you grant me write access to the repository to merge the pull requests? Or if it is not a good idea to grant me write permission to the repository, could someone merge the pull requests on my behalf?
> 
> Also, the gems need to be published to rubygems.org after the merge, if my understanding is not wrong.
